src/background/index.js

```diff
@@ -14,37 +18,37 @@ async function openEnitiyInNewTab(id) {
 function pushProposalToSidebar(proposals, tid) {
 	proposals.fromTab = tid;
 	if (!sidebarLocked) {
-		browser.sidebarAction.setPanel({
-			tabId: tid,
-			panel: browser.runtime.getURL('sidebar/connector.html') + '?' + encodeURIComponent(JSON.stringify(proposals)),
-		});
+		return setSidebarUrl(tid, browser.runtime.getURL('sidebar/connector.html') + '?' + encodeURIComponent(JSON.stringify(proposals)))
 	}
 }
 
-browser.browserAction.onClicked.addListener((tab) => {
+setupCommandListener()
+
+const toggleInlineSidebar = async () =>
+    Browser.sendMessageToActiveTab({type: "toggle-sidebar"})
+
+browser.browserAction.onClicked.addListener(async (tab) => {
 	let tid = tab.id;
 	if (!tabStates[tid]) {
 		tabStates[tid] = {};
 	}
 	if (browser.sidebarAction) {
 		if (!tabStates[tid].sidebarOpen) {
 			if (tabStates[tid].mode === 'show_entity') {
-				(async () => {
-					pushEnitiyToSidebar(tabStates[tid].entity, tid);
-				})();
+				await pushEnitiyToSidebar(tabStates[tid].entity, tid);
 			} else if(tabStates[tid].mode === 'propose_match') {
-				(async () => {
-					pushProposalToSidebar(tabStates[tid].proposals, tid);
-				})();
+                await pushProposalToSidebar(tabStates[tid].proposals, tid)
 			}
-			browser.sidebarAction.open();
+			await browser.sidebarAction.open();
 			tabStates[tid].sidebarOpen = true;
 		} else {
-			browser.sidebarAction.close();
+			await browser.sidebarAction.close();
 			tabStates[tid].sidebarOpen = false;
 		}
 	} else {
-		openEnitiyInNewTab(tabStates[tid].entity);
+        //todo need better handling here if we actually want "open in new tab" behavior in some cases
```

Comment:
@fuddl do we want that behavior? 
I see a few commits on having "an alternative to sidebar" but I'm note entirely sure what was the overall vision there 🤔